InterGalacticCurrencyConvertor/src/main/java/materials/MaterialFactory.java

```diff
@@ -0,0 +1,70 @@
+package main.java.materials;
+
+import java.util.HashMap;
+import java.util.Objects;
+
+import main.java.IgnConversionErrors;
+import main.java.IgnConversionException;
+
+/**
+ * Keeps track of different types of materials that can be sold InterGalactically.
+ */
+public final class MaterialFactory {
+  private final HashMap<String, Material> mMtlMap  = new HashMap<>();
+  private final static MaterialFactory mInstance = new MaterialFactory();
+
+  public static MaterialFactory getInstance() {
+    return mInstance;
+  }
+
+  public void add(String name, int numUnits, double totalCost)
+      throws IgnConversionException {
+    Objects.requireNonNull(name, "Invalid Material Name");
```

Comment:
Can name be empty? If not, validate against that as well.

Reply:
done in Material class but this will not occur as user input will have a valid material name